CRM/Membershipextrasdata/Upgrader.php

```diff
@@ -360,4 +361,22 @@ private function enableDiscountFieldsOnWebforms() {
     }
   }
 
+  /**
+   * Set default manual direct debit configurations
+   */
+  private function setDefaultManualDirectDebitConfigurations() {
+    $configFields = [
+      'manualdirectdebit_default_reference_prefix' => 'DD',
+      'manualdirectdebit_minimum_reference_prefix_length' => 6,
+      'manualdirectdebit_new_instruction_run_dates' => [1],
+      'manualdirectdebit_payment_collection_run_dates' => [1],
+      'manualdirectdebit_minimum_days_to_first_payment' => 2,
+      'manualdirectdebit_days_in_advance_for_collection_reminder' => 3,
+      'manualdirectdebit_batch_submission_queue_limit' => 50,
+    ];
+    foreach ($configFields as $name => $value) {
+      civicrm_api3('setting', 'create', [$name => $value]);
```

Comment:
@ahed-compucorp  Would be better if we don't need foreach? We just `civicrm_api3('setting', 'create', $configFields); `

Reply:
I tried that but It fired an error 

    [is_error] => 1
    [error_message] => manualdirectdebit_default_reference_prefix,manualdirectdebit_minimum_reference_prefix_length,manualdirectdebit_new_instruction_run_dates,manualdirectdebit_payment_collection_run_dates,manualdirectdebit_minimum_days_to_first_payment,manualdirectdebit_days_in_advance_for_collection_reminder,manualdirectdebit_batch_submission_queue_limit not valid settings

Then I used `Civi::settings()->add` instead and it works.

Reply:
it does not work because "Your setting will not be active until you run flush all caches (e.g cv flush or drush cvapi System.flush)"  https://docs.civicrm.org/dev/en/latest/framework/setting/#adding-a-new-setting-to-civicrm-core but using Civi::settings()->add is okay too.